easybuild/framework/easyconfig/tweak.py

```diff
@@ -120,7 +120,17 @@ def tweak(easyconfigs, build_specs, modtool, targetdirs=None):
                 target_toolchain['version'] = source_toolchain['version']
 
             src_to_dst_tc_mapping = map_toolchain_hierarchies(source_toolchain, target_toolchain, modtool)
+            if src_to_dst_tc_mapping is None:
+                if not build_option('force'):
+                    raise EasyBuildError("Toolchain %s is not equivalent to toolchain %s in terms of capabilities. "
+                                         "Use --force if you are sure.",
+                                         target_toolchain['name'], source_toolchain['name'])
+                print_warning("Combining --try-toolchain with --force for toolchains with unequal capabilities: "
+                              "using regex.")
+                revert_to_regex = True
+                modifying_toolchains = False
```

Comment:
@bartoldeman Please refactor this a bit to i) use positive logic, ii) use `if/else` (it's a bit confusing now imho).

I would also clarify the "regex" part a bit in the warning, and also the error message(see below).

```python
if build_option('force'):
    print_warning("Combining --try-toolchain with --force for toolchains with unequal capabilities: "
                  "disabling recursion & not changing (sub)toolchains.")
    revert_to_regex = True
    modifying_toolchains = False
else:
    raise EasyBuildError("Toolchain %s is not equivalent to toolchain %s in terms of capabilities, aborting "
                         "(if you know what you are doing, you can use --force to proceed anyway)",
                         target_toolchain['name'], source_toolchain['name'])

Reply:
This is not quite the same thing. Here you are saying if we have `force` then asssume we want regex, whereas the other approach is to only go looking for force if the mapping was not possible.